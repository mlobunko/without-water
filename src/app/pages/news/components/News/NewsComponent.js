import React from 'react';
import PropTypes from 'prop-types';

import Alert from 'shared/error/components/Alert';
import Loader from 'shared/fetching/components/Loader';
import HeaderH1 from 'app/header/components/HeaderH1';

import Articles from '../Articles';
import { StyledNewsComponent } from './styles';

export const NewsComponent = ({ fetching, isError, news, serverError }) => (
  <StyledNewsComponent>
    <div>
      <HeaderH1 text={'News'} />
    </div>
    {!news.length && fetching ? (
      <Loader centerPage />
    ) : !isError ? (
      <Articles />
    ) : (
      <Alert color="danger" textError={serverError} centerPage />
    )}
  </StyledNewsComponent>
);
NewsComponent.displayName = 'NewsComponent';

export default NewsComponent;

NewsComponent.propTypes = {
  fetching: PropTypes.bool.isRequired,
  isError: PropTypes.bool.isRequired,
  news: PropTypes.arrayOf(
    PropTypes.shape({
      title: PropTypes.string.isRequired,
      text: PropTypes.string.isRequired
    })
  ),
  serverError: PropTypes.string
};
