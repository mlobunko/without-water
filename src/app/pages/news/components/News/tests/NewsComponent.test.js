import React from 'react';
import { shallow } from 'enzyme';

import { NewsComponent } from '../NewsComponent';

describe('render', () => {
  const initialProps = {
    fetching: true,
    isError: true,
    news: [
      {
        title: '123',
        text: 'hola'
      }
    ],
    serverError: 'textError'
  };

  it('renders Loader', () => {
    const props = {
      ...initialProps,
      isError: false,
      news: [],
      serverError: ''
    };
    const wrapper = shallow(<NewsComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Articles', () => {
    const props = {
      ...initialProps,
      isError: false,
      serverError: '',
      fetching: false
    };
    const wrapper = shallow(<NewsComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('renders Alert', () => {
    const props = {
      ...initialProps,
      news: [],
      fetching: false
    };
    const wrapper = shallow(<NewsComponent {...props} />);
    expect(wrapper).toMatchSnapshot();
  });
});
