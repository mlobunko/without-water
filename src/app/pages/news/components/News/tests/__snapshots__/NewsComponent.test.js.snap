// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render renders Alert 1`] = `
<styled.div>
  <div>
    <HeaderH1
      text="News"
    />
  </div>
  <Alert
    centerPage={true}
    color="danger"
    textError="textError"
  />
</styled.div>
`;

exports[`render renders Articles 1`] = `
<styled.div>
  <div>
    <HeaderH1
      text="News"
    />
  </div>
  <Connect(Component) />
</styled.div>
`;

exports[`render renders Loader 1`] = `
<styled.div>
  <div>
    <HeaderH1
      text="News"
    />
  </div>
  <Component
    centerPage={true}
  />
</styled.div>
`;
